lib/query/lib/prepareForProcess.js

```diff
@@ -44,12 +44,40 @@ function applyPagination(body, _params) {
     }
 }
 
-export default (_body, _params = {}) => {
+function handleTextFilters(data, isClient) {
+    _.each(data, field => {
+        if (field.$text) {
+            if (isClient) {
+                let $or = _.map(field.$text.$fields, (val, key) => {
+                    return {
+                        [key]: {
+                            $regex:field.$text.$search,
+                            $options: field.$text.$caseSensitive ? '' : 'i'
+                        }
+                    }
+                })
+                delete field.$text
+                if (_.isEmpty(field)) {
+                    field = {$or}
```

Comment:
Oh right, I always forget that you can't do field:{$or:[]} in Mongo.
It seemed to work in my testing though... It must have been that the $text filter on the server was working, and the client side filtering only matters when you have multiple subscriptions on the same collection.